csharp/src/Apache.Arrow/Types/ListType.cs

```diff
@@ -25,10 +25,16 @@ public sealed class ListType : ArrowType
         public Field ValueField { get; }
         public IArrowType ValueDataType { get; }
 
-        public ListType(Field valueField, IArrowType valueDataType)
+        public ListType(Field valueField)
         {
             ValueField = valueField ?? throw new ArgumentNullException(nameof(valueField));
+            ValueDataType = ValueField.DataType;
+        }
+
+        public ListType(IArrowType valueDataType)
+        {
             ValueDataType = valueDataType ?? NullType.Default;
```

Comment:
Should we throw an exception for `null` `valueDataType`?
`NullType` isn't usable.

@eerhardt What do you think about this?

Reply:
The line was removed by the new commit but same check exists in Field.
Should we throw an exception there?

https://github.com/apache/arrow/blob/master/csharp/src/Apache.Arrow/Field.cs#L44

Reply:
I don't see this behavior in either the C++ or Java implementation.

I'd say it is up to you whether to leave the Field constructor as-is, or change it to throw on a `null` type. I could go either way on it.

Reply:
OK. We can do it as a follow-up task when we want to do.
We can merge this.